tutorial3_distributions/demo/main.js

```diff
@@ -1,133 +1,127 @@
-   *    new Set([1, 2, 1, 4, 5, 6, 4, 6, 10])
-   * and:
-   *    [...new Set([1, 2, 1, 4, 5, 6, 4, 6, 10])]
-   */
-  const uniqueStates = [...new Set(data.map(d => d.state))];
-  // add dropdown for interaction
-  // select reference: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select
+  // add dropdown (HTML selection) for interaction
+  // HTML select reference: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select
   const selectElement = d3.select("#dropdown").on("change", function() {
-    console.log("new selected state is", this.value);
+    console.log("new selected party is", this.value);
     // `this` === the selectElement
     // this.value holds the dropdown value a user just selected
-    selectedState = this.value;
+    state.selectedParty = this.value;
     draw(); // re-draw the graph based on this new selection
   });
 
   // add in dropdown values from the unique options in the data
   selectElement
     .selectAll("option")
-    .data(uniqueStates)
+    .data(["D", "R", "Both"])
     .join("option")
     .attr("value", d => d)
-    .text(d => d);
+    .text(d => d)
 
   // create an svg element in our main `d3-container` element
   svg = d3
     .select("#d3-container")
     .append("svg")
-    .attr("width", width)
-    .attr("height", height);
+    .attr("width", constants.width)
+    .attr("height", constants.height);
 
   // add the xAxis
   svg
     .append("g")
     .attr("class", "x-axis")
-    .attr("transform", `translate(0,${height - margin.bottom})`)
+    .attr("transform", `translate(0,${constants.height - constants.margin.bottom})`)
     .call(xAxis);
 
   // add the xAxis
   svg
     .append("g")
     .attr("class", "y-axis")
-    .attr("transform", `translate(${margin.left},0)`)
+    .attr("transform", `translate(${constants.margin.left},0)`)
     .call(yAxis);
 
   draw();
 }
 
+// DRAW FUNCTION
 function draw() {
   // we call this everytime there is an update to the data/state
 
-  // filter the data for the selectedState
-  let filteredData = data;
-  // if there is a selectedState, filter the data before drawing it
-  if (selectedState) {
-    filteredData = data.filter(d => d.state === selectedState);
+  // filter the data for the selectedParty
+  let filteredData = state.data;
+  // if there is a selectedParty, filter the data before drawing it
+  if (state.selectedParty) {
+    filteredData = state.data.filter(d => 
+      state.selectedParty === "Both" 
+    || d.party === state.selectedParty);
+    console.log("new filtered data includes", filteredData.length, "politicians");
   }
 
-  /** create a `g`, or group element
-   * This `g` will contain within it the `circle` and `text` elements
-   * We do it this way so that we don't have to position the `circle` and `text` separtely
-   * (since we want to keep the label tied to the circle it is annotating)
-   */
-  //
-  const dotgroup = svg
+  const dot = svg
```

Comment:
I moved from a `group` to just a `circle` for a few reasons:

1. I find it easier to see that `cx`, `cy`, and `r` are all attributes of a circle, so it makes sense to include them directly after the `append('circle')`, in my opinion
2. We have to use `.call` on the transitions anyway -- so I thought it was simpler to introduce its functionality here, in this context, than the way we discussed last week

This is flexible too -- but you mentioned you wanted to see how I would approach it so I doubled down. We can discuss tomorrow!